### DISCUSSION TITLE
Cloudflare Workers Not Supporting Supabase JS Client

### DISCUSSION THREAD

**Discussion by nascode** [supabase/supabase#588] [Questions] [answered]
Hi there, 

I want to try accessing Supabase db via Cloudflare workers (CFW). However CFW doesnt support XMLHTTPRequest. CFW only support fetch.

Digging deeper, it seems that cross-fetch package who called the actual XHR. 

Is there any recommendation to solve this?

**Comment by inian** [comment]
cross-fetch [doesn't work](https://github.com/lquixada/cross-fetch/issues/69) in a Cloudflare worker. Looks like faunadb-js also had [this issue](https://community.cloudflare.com/t/xmlhttprequest-is-not-defined/133866/3) and they worked around it by allowing users to pass in a custom fetch implementation in the constructor..that seems hacky though.

**Comment by nascode** [comment]
Since from what i know it's not possible to connect with tcp in bare Cloudflare Workers, using REST api is CFW way to communicate with database. 

I think Supabase is using POSTGREST already, i hope we can have option similar with fauna to pass in custom fetch implementation on Subase js client.

**Reply by kiwicopple** [reply]
I think it would be best to support cross-fetch by fixing [this](https://github.com/lquixada/cross-fetch/issues/69), rather than engineering the lib for an optional fetch provider.

Did you try the workaround in the issue? It's pretty hacky, but seems like it could work immediately

**Comment by nascode** [comment, marked as answer]
Hi @kiwicopple @inian 

I successfully use `patch-package` to patch `cross-fetch` so it use global `fetch` instead.

I made a repo for this at https://github.com/nascode/cloudflare-workers-supabase-fix

I hope this helps people who get stuck with similar problem.

**Reply by kiwicopple** [reply]
Hi @nascode - thanks for this repo, I've already shared it with a couple of people using Cloudflare + Supabase. Would you be mind if I copy this to our [examples](https://github.com/supabase/supabase/tree/master/examples) folder? I'll give credit in the readme and link to your repo

**Reply by nascode** [reply]
Hi @kiwicopple 

Of course, feel free to do that. Cheers!

**Comment by kristianfreeman** [comment]
Dropping in another solution here - as @munster mentioned [here](https://github.com/lquixada/cross-fetch/issues/69#issuecomment-804210923), you can use Webpack's `externals` feature to patch `cross-fetch` directly and start to work with Supabase's JS client directly inside of a Cloudflare Workers function:

```js
// webpack.config.js
module.exports = {
  externals: [{ 'cross-fetch': 'fetch' }]
}
```

I made an example project here to show off what that might look like - note that stuff like realtime and auth isn't tested yet, though I'd like to give it a shot soon: https://github.com/signalnerve/supabase-workers-proxy

**Reply by inian** [reply]
This seems cleaner @signalnerve. Thanks for sharing your solution :)

**Reply by nascode** [reply]
This is great! TIL. Would be a great addition on Supabase docs.

**Reply by inian** [reply]
Good idea. Added your repo to our examples page @signalnerve https://supabase.io/docs/guides/examples
